backend/inspirehep/records/api/mixins.py

```diff
@@ -306,160 +277,91 @@ def _download_file_from_url(self, url):
+
         Returns:
             str: hash for copied file
+
         Raises:
             ValueError: when hash to verify was provided and
                 it didn't match to file hash
-
         """
         if original_key:
-            key = self._verify_file(file, original_key)
+            key = self.verify_file(file_, original_key)
             if not key:
                 raise ValueError("File verifiaction failed! Hashes do not match!")
         else:  # Compatibility with old way of holding files
-            key = self.hash_data(file_instance=file)
+            key = self.hash_data(file_instance=file_)
+
         if key not in self.files.keys:
             LOGGER.debug("Adding file to record", key=key, uuid=self.id)
-            file.copy(bucket=self.files.bucket.id, key=key)
+            file_.copy(bucket=self.files.bucket.id, key=key)
             self["_files"] = self.files.dumps()
         else:
             LOGGER.debug("File already attached to record", key=key, uuid=self.id)
         return key
 
-    def get_bucket(self, location=None, storage_class=None, record_id=None):
-        """Allows to retrieve bucket for any record(default: self)
-
-        Args:
-            location (str): Bucket location
-                (default: 'RECORDS_DEFAULT_FILE_LOCATION_NAME') from config
-            storage_class (str): Bucket storage class
-                (default: 'RECORDS_DEFAULT_STORAGE_CLASS') from config
-            record_id (int): record to which bucket is asigned
-                (default: self)
-        Returns:
-            Bucket: if found in db for selected location, storage_class and record_id or
-                None if there were no bucket.
-        Raises:
-            NoResultFound: When provided location was not found.
-
-        """
-        if not storage_class:
-            storage_class = current_app.config["RECORDS_DEFAULT_STORAGE_CLASS"]
-        if not location:
-            location = current_app.config["RECORDS_DEFAULT_FILE_LOCATION_NAME"]
-        if not record_id:
-            record_id = self.id
-        try:
-            location_obj = Location.get_by_name(location)
-        except NoResultFound:
-            raise NoResultFound(
-                "Cannot find location %s. Please check if system is configured properly!",
-                location,
-            )
-
-        bucket = (
-            RecordsBuckets.query.join(Bucket)
-            .filter(
-                RecordsBuckets.record_id == record_id,
-                Bucket.default_storage_class == storage_class,
-                Bucket.default_location == location_obj.id,
-            )
-            .one_or_none()
-        )
-        if bucket:
-            LOGGER.debug("Bucket found", key=bucket.bucket.id, uuid=self.id)
-            return bucket.bucket
-        LOGGER.info("Bucket not found", uuid=self.id)
-        return self._create_bucket(location, storage_class)
-
-    def _create_bucket(self, location=None, storage_class=None):
```

Comment:
They are still saying to implement your own _create_bucket
https://github.com/inveniosoftware/invenio-records-files/blob/ad9bd1dfbd02eb745e17d2cc23a1dc3583e6dbcc/invenio_records_files/api.py#L242-L243

And actually their create_bucket method is just silly...

Reply:
This is the subclass :) https://github.com/inveniosoftware/invenio-records-files/blob/ad9bd1dfbd02eb745e17d2cc23a1dc3583e6dbcc/invenio_records_files/api.py#L325